PyInstaller/building/api.py

```diff
@@ -354,6 +354,9 @@ def __init__(self, *args, **kwargs):
         else:
             self.upx = False
 
+        # Save the console value for later use on macOS
+        CONF['console'] = self.console
```

Comment:
No need for this. The `console` can be picked up in `EXE` at https://github.com/pyinstaller/pyinstaller/pull/3566/files#diff-c335c5207e1c5d707f757c131b913b62L62. And when doing this, `console` can be per EXE, while when storing the value in `CONFIG` it's global for all executable.

Reply:
I would prefer too, but condition `if isinstance(arg, EXE):` is never taken.

Reply:
IC. According to the templates (PyInstaller/building/templates.py), only for a one-file BUNDLE, the first argument is an EXE. For the one-dir Bundle, it is a COLLECT.

Two possible solutions:
1) In COLLECT there is already a loop scanning for EXE. Use this to set COLLECT's `.console` for the collection. This has the drawback, that we hide "a bundle can contain only one exe" in an unrelated area of the code.
2) In BUNDLE, if it is a COLLECT, scan the COLLECT's `.toc` for an exe and fetch the info from there. I just discover that [there is already a loop searching for Exe](https://github.com/BoboTiG/pyinstaller/blob/v3.3.1/PyInstaller/building/osx.py#L78). This loop takes into account the first exe only, thus I would do the same for `.console`.

Reply:
The option 2 is not doable (as of now) because data in `toc` is just strings, for instance something like:
```python
('systray', '.../systray/build/systray/systray', 'EXECUTABLE')
```
If we have had an attribute pointing to the `EXE` object it would have been easy. But I do not think it is a good thing to do that.

I updated the PR to remove the use of `CONF` using option 1.